keps/sig-network/20190415-Autopath API for clusterDNS.md

```diff
@@ -0,0 +1,100 @@
+---
+title: DNS Autopath in PodSpec
+authors:
+  - "@prameshj"
+owning-sig: sig-network
+participating-sigs:
+  - sig-network
+reviewers:
+  - "@thockin"
+  - "@bowei"
+  - "@johnbelamaric"
+approvers:
+  - "@thockin"
+  - "@bowei"
+  - "@johnbelamaric"
+creation-date: 2019-04-15
+last-updated: 2019-04-15
+status: provisional
+---
+# DNS Autopath in PodSpec
+
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Implementation History](#implementation-history)
+* [Alternatives [optional]](#alternatives-optional)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+This proposal aims to minimize the number of parallel DNS queries generated by a pod by moving searchpath expansion logic to the DNS server side. This introduces a new dnsPolicy that can be configured on a per-pod basis. The metadata required to complete the searchpath expansion will be sent to the DNS Server via an EDNS0 option. This will be inserted into the request by the Nodelocal DNSCache.
```

Comment:
Added this. They also seem to be parallel in glibc since 2.16.
https://abi-laboratory.pro/?view=changelog&l=glibc&v=2.16.0

DNS IPv4-IPv6 parallel lookup now deals better with broken DNS
  servers (the case, e.g., for some people using the built-in DNS
  server in ADSL modems/routers).  There is a once-per-process timeout
  in case of a broken server.  To avoid it, users can run nscd or put
  'options single-request' in /etc/resolv.conf.

Reply:
I believe that is describing that IPv4 and IPv6 request are done in parallel. IIRC, in musl search path requests are done in parallel as well. So the reduction in parallel queries would only be for musl clients. I think the reduction in client latency and query load in glibc case is just as compelling.